<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jean Morrison" />

<meta name="date" content="2017-04-10" />

<title>Analysis of CAD and LDL</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sherlock2 model using Ash</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Analysis of CAD and LDL</h1>
<h4 class="author"><em>Jean Morrison</em></h4>
<h4 class="date"><em>April 10, 2017</em></h4>

</div>


<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In this analysis I look for a signature of a causal relationship between LDL cholesterol and coronary artery disease. This analysis was inspired by the positive results reported by Pickrell et al. in “Detection and interpretation of shared genetic influences on 40 human traits” and we use the same data.</p>
<p>Data sources: + CAD summary statistics were downloaded from <a href="http://www.cardiogramplusc4d.org/data-downloads/">here</a> following the “CARDIoGRAM GWA meta-analysis” link. + LDL summary statistics were downloaded from <a href="http://csg.sph.umich.edu/abecasis/public/lipids2010/">here</a></p>
</div>
<div id="data-processing-and-summaries" class="section level2">
<h2>Data processing and summaries</h2>
<p>Coronary artery disease:</p>
<pre class="r"><code>library(readr)
cad &lt;- read_delim(&quot;../data/CARDIoGRAM_GWAS_RESULTS.txt&quot;, delim=&quot;\t&quot;)
head(cad)
# A tibble: 6 × 12
#         SNP `chr_pos_(b36)` reference_allele other_allele ref_allele_frequency    pvalue het_pvalue   log_odds log_odds_se N_case
#       &lt;chr&gt;           &lt;chr&gt;            &lt;chr&gt;        &lt;chr&gt;                &lt;dbl&gt;     &lt;dbl&gt;      &lt;dbl&gt;      &lt;dbl&gt;       &lt;dbl&gt;  &lt;int&gt;
#1 rs12565286     chr1:711153                C            G           0.05380729 0.0651210  0.9996689  0.1282012   0.0695073   6659
#2 rs11804171     chr1:713682                T            A           0.94596855 0.0632817  0.9992190 -0.1297868   0.0698828   6017
#3  rs3094315     chr1:742429                G            A           0.17509983 0.9545154  0.9934594  0.0016609   0.0291187   9714
#4  rs3131968     chr1:744055                G            A           0.76986967 0.3571689  0.5146446 -0.0289617   0.0314537  10213#
#5  rs2905035     chr1:765522                G            A           0.81987365 0.8823145  0.5314654 -0.0040285   0.0272131  13523
#6  rs2980319     chr1:766985                T            A           0.82053910 0.9730835  0.6074723 -0.0009125   0.0270433  14201
# ... with 2 more variables: N_control &lt;int&gt;, model &lt;chr&gt;
dim(cad)
#[1] 2420360      12</code></pre>
<p>LDL cholesterol:</p>
<pre class="r"><code>ldl &lt;- read_delim(&quot;../data/LDL_with_Effect.tbl&quot;, delim=&quot;\t&quot;)
dim(ldl)
#[1] 2692564      10
head(ldl)
# A tibble: 6 × 10
#  MarkerName Allele1 Allele2 Weight GC.Zscore GC.Pvalue Overall                 Direction  Effect StdErr
#       &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;int&gt;     &lt;dbl&gt;     &lt;dbl&gt;   &lt;chr&gt;                     &lt;chr&gt;   &lt;dbl&gt;  &lt;dbl&gt;
#1       rs10       a       c  81680     2.051   0.04027       + ++-+++++-+?++-??-++-??++-  0.0294 0.0152
#2  rs1000000       a       g  95454     0.662   0.50770       + -----++-+--+-+++-+++-+-++  0.0044 0.0063
#3 rs10000010       t       c  95454     1.583   0.11350       + ++--++---+-+-++--+++---++  0.0073 0.0052
#4 rs10000012       c       g  95397     0.155   0.87700       + +-+-----++-++---+++---+-+ -0.0002 0.0076
#5 rs10000013       a       c  95454    -1.392   0.16400       - --+---+--++--+--++---+-++ -0.0075 0.0062
#6 rs10000017       t       c  95454    -0.015   0.98820       - --+---+++--+---+-++-++--+  0.0021 0.0066</code></pre>
<p>Merge data into a single data frame</p>
<pre class="r"><code>sum(cad$SNP %in% ldl$MarkerName)
#[1] 2420288
ix &lt;- which(cad$SNP %in% ldl$MarkerName)
dat &lt;- data.frame(cad[ix, c(&quot;SNP&quot;, &quot;chr_pos_(b36)&quot;, &quot;reference_allele&quot;, &quot;other_allele&quot;, &quot;log_odds&quot;, &quot;log_odds_se&quot;, &quot;pvalue&quot;)])
names(dat)[5:7] &lt;- c(&quot;cad_log_odds&quot;, &quot;cad_log_odds_se&quot;, &quot;cad_pvalue&quot;)
dat$ldl_effect &lt;- ldl$Effect[match(dat$SNP, ldl$MarkerName)]
dat$ldl_se &lt;- ldl$StdErr[match(dat$SNP, ldl$MarkerName)]
dat$ldl_pvalue &lt;- ldl$GC.Pvalue[match(dat$SNP, ldl$MarkerName)]

#Check Strands
letters &lt;- c(&quot;a&quot;=&quot;A&quot;, &quot;c&quot;=&quot;C&quot;, &quot;g&quot;=&quot;G&quot;, &quot;t&quot;=&quot;T&quot;)
ix_match &lt;- which(letters[ldl$Allele1[match(dat$SNP, ldl$MarkerName)]] == dat$reference_allele &amp; letters[ldl$Allele2[match(dat$SNP, ldl$MarkerName)]] == dat$other_allele   )
length(ix_match)
#[1] 212913
ix_mismatch &lt;- which(letters[ldl$Allele2[match(dat$SNP, ldl$MarkerName)]] == dat$reference_allele &amp; letters[ldl$Allele1[match(dat$SNP, ldl$MarkerName)]] == dat$other_allele)
length(ix_mismatch)
#[1] 2207284
ix_other &lt;- which(!1:nrow(dat) %in% c(ix_match, ix_mismatch))
length(ix_other)
#[1] 91
dat$ldl_effect[ix_mismatch] &lt;- -1*dat$ldl_effect[ix_mismatch]
dat &lt;- dat[-ix_other,]
nrow(dat)
#[1] 2420197

save(dat, file=&quot;../data/LDL_CAD.RData&quot;)</code></pre>
</div>
<div id="basic-plots-approximately-replicating-pickerell-et-al-observations" class="section level2">
<h2>Basic plots (approximately) replicating Pickerell et al observations</h2>
<p>First look at effect sizes ascertaining based on LDL and ascertaining based on CAD. Note that, in these plots, we haven’t pruned for LD. Results after pruning are shown further on.</p>
<pre class="r"><code>library(readr)
library(sherlockAsh)</code></pre>
<pre><code>## Warning: replacing previous import &#39;ks::compare&#39; by &#39;loo::compare&#39; when
## loading &#39;sherlockAsh&#39;</code></pre>
<pre class="r"><code>dat &lt;- getobj(&quot;../data/LDL_CAD.RData&quot;)
dat$cad_qvalue &lt;- p.adjust(dat$cad_pvalue, method=&quot;BH&quot;)
with(dat[dat$cad_qvalue &lt; 0.1,], plot(ldl_effect, cad_log_odds, cex=0.5, main=&quot;Only top CAD SNPs&quot;, xlab=&quot;LDL effect&quot;, ylab=&quot;CAD Log Odds&quot;))</code></pre>
<p><img src="cad_ldl_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<pre class="r"><code>dat$ldl_qvalue &lt;- p.adjust(dat$ldl_pvalue, method=&quot;BH&quot;)
with(dat[dat$ldl_qvalue &lt; 0.1,], plot(ldl_effect, cad_log_odds, cex=0.5, main=&quot;Only top LDL SNPs&quot;, xlab=&quot;LDL effect&quot;, ylab=&quot;CAD Log Odds&quot;))</code></pre>
<p><img src="cad_ldl_files/figure-html/unnamed-chunk-4-2.png" width="672" /></p>
<p>Visually, these traits appear to have patterns consistent with a causal effect of LDL on CAD and not constent with a causal effect of CAD on LDL. We see that among the top LDL SNPs, there is a strong correlation between LDL effect size and CAD effect size. However, reversing the procedure and selecting the top CAD SNPs, the trend is les spronounced. Some of the top CAD SNPs also have large LDL effect sizes, however there are many strong CAD SNPs that have LDL effect close to zero.</p>
<p>Within the set of top LDL SNPs, there is correlation between LDL effect and CAD effect even for SNPs that do not reach the significance threshold for CAD:</p>
<pre class="r"><code>cat(&quot;Correlation of effect sizes among all top LDL SNPs: &quot;, with(dat[dat$ldl_qvalue &lt; 0.1,], cor(ldl_effect, cad_log_odds)), &quot;\n&quot;)</code></pre>
<pre><code>## Correlation of effect sizes among all top LDL SNPs:  0.5311864</code></pre>
<pre class="r"><code>cat(&quot;Correlation for significant LDL SNPs that have weak CAD effect: &quot;, with(dat[dat$ldl_qvalue &lt; 0.1 &amp; dat$cad_qvalue &gt; 0.1,], cor(ldl_effect, cad_log_odds)), &quot;\n&quot;)</code></pre>
<pre><code>## Correlation for significant LDL SNPs that have weak CAD effect:  0.5276117</code></pre>
<pre class="r"><code>cat(&quot;Correlation for significant LDL SNPs that have strong CAD effect: &quot;, with(dat[dat$ldl_qvalue &lt; 0.1 &amp; dat$cad_qvalue &lt; 0.1,], cor(ldl_effect, cad_log_odds)), &quot;\n&quot;)</code></pre>
<pre><code>## Correlation for significant LDL SNPs that have strong CAD effect:  0.6966864</code></pre>
<p>However, among top CAD SNPs, there is only (positive) correlation in effect sizes for SNPs that are also strong LDL SNPs</p>
<pre class="r"><code>cat(&quot;Correlation of effect sizes among all top CAD SNPs: &quot;, with(dat[dat$cad_qvalue &lt; 0.1,], cor(ldl_effect, cad_log_odds)), &quot;\n&quot;)</code></pre>
<pre><code>## Correlation of effect sizes among all top CAD SNPs:  0.0788183</code></pre>
<pre class="r"><code>cat(&quot;Correlation for significant CAD SNPs that have weak LDL effect: &quot;, with(dat[dat$cad_qvalue &lt; 0.1 &amp; dat$ldl_qvalue &gt; 0.1,], cor(ldl_effect, cad_log_odds)), &quot;\n&quot;)</code></pre>
<pre><code>## Correlation for significant CAD SNPs that have weak LDL effect:  -0.216885</code></pre>
<pre class="r"><code>cat(&quot;Correlation for significant CAD SNPs that have strong LDL effect: &quot;, with(dat[dat$cad_qvalue &lt; 0.1 &amp; dat$ldl_qvalue &lt; 0.1,], cor(ldl_effect, cad_log_odds)), &quot;\n&quot;)</code></pre>
<pre><code>## Correlation for significant CAD SNPs that have strong LDL effect:  0.6966864</code></pre>
<p>This suggests that the data are more consistant with a causal effect of LDL on CAD than vice versa.</p>
</div>
<div id="height-and-ldl" class="section level2">
<h2>Height and LDL</h2>
<p>As far as I know, there are no reports of a strong link between adult height and lipid levels. I am hoping that this can provide a negative example for us. Height data were downloaded from <a href="http://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files#GWAS_Anthropometric_2014_Height">here</a>.</p>
<p>LDL cholesterol:</p>
<pre class="r"><code>ldl &lt;- read_delim(&quot;../data/LDL_with_Effect.tbl&quot;, delim=&quot;\t&quot;)
dim(ldl)
#[1] 2692564      10
head(ldl)
# A tibble: 6 × 10
#  MarkerName Allele1 Allele2 Weight GC.Zscore GC.Pvalue Overall                 Direction  Effect StdErr
#       &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;int&gt;     &lt;dbl&gt;     &lt;dbl&gt;   &lt;chr&gt;                     &lt;chr&gt;   &lt;dbl&gt;  &lt;dbl&gt;
#1       rs10       a       c  81680     2.051   0.04027       + ++-+++++-+?++-??-++-??++-  0.0294 0.0152
#2  rs1000000       a       g  95454     0.662   0.50770       + -----++-+--+-+++-+++-+-++  0.0044 0.0063
#3 rs10000010       t       c  95454     1.583   0.11350       + ++--++---+-+-++--+++---++  0.0073 0.0052
#4 rs10000012       c       g  95397     0.155   0.87700       + +-+-----++-++---+++---+-+ -0.0002 0.0076
#5 rs10000013       a       c  95454    -1.392   0.16400       - --+---+--++--+--++---+-++ -0.0075 0.0062
#6 rs10000017       t       c  95454    -0.015   0.98820       - --+---+++--+---+-++-++--+  0.0021 0.0066</code></pre>
<p>Height</p>
<pre class="r"><code>ht &lt;- read_delim(&quot;../data/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq.txt&quot;, delim=&quot;\t&quot;)
dim(ht)
[1] 2550858       8
head(ldl)
# A tibble: 6 × 8
#  MarkerName Allele1 Allele2 Freq.Allele1.HapMapCEU       b     SE       p      N
#       &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;                  &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;  &lt;int&gt;
#1  rs4747841       A       G                  0.551 -0.0011 0.0029 7.0e-01 253213
#2  rs4749917       T       C                  0.436  0.0011 0.0029 7.0e-01 253213
#3   rs737656       A       G                  0.367 -0.0062 0.0030 4.2e-02 253116
#4   rs737657       A       G                  0.358 -0.0062 0.0030 4.1e-02 252156
#5  rs7086391       T       C                  0.120 -0.0087 0.0038 2.4e-02 248425
#6   rs878177       T       C                  0.300  0.0140 0.0031 8.2e-06 251271</code></pre>
<p>Merge data into a single data frame</p>
<pre class="r"><code>sum(ht$MarkerName %in% ldl$MarkerName)
#[1] 2535734
ix &lt;- which(ht$MarkerName %in% ldl$MarkerName)
dat &lt;- data.frame(ht[ix, c(&quot;MarkerName&quot;, &quot;Allele1&quot;, &quot;Allele2&quot;, &quot;b&quot;, &quot;SE&quot;, &quot;p&quot;)])
names(dat)[4:6] &lt;- c(&quot;height_effect&quot;, &quot;height_se&quot;, &quot;height_pvalue&quot;)
dat$ldl_effect &lt;- ldl$Effect[match(dat$MarkerName, ldl$MarkerName)]
dat$ldl_se &lt;- ldl$StdErr[match(dat$MarkerName, ldl$MarkerName)]
dat$ldl_pvalue &lt;- ldl$GC.Pvalue[match(dat$MarkerName, ldl$MarkerName)]

#Check Strands
letters &lt;- c(&quot;a&quot;=&quot;A&quot;, &quot;c&quot;=&quot;C&quot;, &quot;g&quot;=&quot;G&quot;, &quot;t&quot;=&quot;T&quot;)
ix_match &lt;- which(letters[ldl$Allele1[match(dat$MarkerName, ldl$MarkerName)]] == dat$Allele1 &amp; 
                  letters[ldl$Allele2[match(dat$MarkerName, ldl$MarkerName)]] == dat$Allele2)
length(ix_match)
#[1] 2534065
ix_mismatch &lt;- which(letters[ldl$Allele2[match(dat$MarkerName, ldl$MarkerName)]] == dat$Allele1 &amp; 
                  letters[ldl$Allele1[match(dat$MarkerName, ldl$MarkerName)]] == dat$Allele2)
length(ix_mismatch)
#[1] 0

ix_other &lt;- which(!1:nrow(dat) %in% c(ix_match, ix_mismatch))
length(ix_other) #These seem to be mostly flipped strands
#[1] 1669

#dat$ldl_effect[ix_mismatch] &lt;- -1*dat$ldl_effect[ix_mismatch]
dat &lt;- dat[-ix_other,]
nrow(dat)
save(dat, file=&quot;../data/LDL_HEIGHT.RData&quot;)</code></pre>
<p>Some basic plots</p>
<pre class="r"><code>library(readr)
library(sherlockAsh)
dat &lt;- getobj(&quot;../data/LDL_HEIGHT.RData&quot;)
dat$height_qvalue &lt;- p.adjust(dat$height_pvalue, method=&quot;BH&quot;)
with(dat[dat$height_qvalue &lt; 0.1,], plot(ldl_effect, height_effect, cex=0.5, main=&quot;Only top Height SNPs&quot;, xlab=&quot;LDL effect&quot;, ylab=&quot;Height effect&quot;))</code></pre>
<p><img src="cad_ldl_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<pre class="r"><code>with(dat[dat$height_qvalue &lt; 0.1,], plot(ldl_effect, height_effect, cex=0.5, 
      main=&quot;Only top Height SNPs&quot;, xlab=&quot;LDL effect&quot;, ylab=&quot;Height effect&quot;, xlim=c(-0.1, 0.1)))</code></pre>
<p><img src="cad_ldl_files/figure-html/unnamed-chunk-10-2.png" width="672" /></p>
<pre class="r"><code>dat$ldl_qvalue &lt;- p.adjust(dat$ldl_pvalue, method=&quot;BH&quot;)
with(dat[dat$ldl_qvalue &lt; 0.1,], plot(ldl_effect, height_effect, cex=0.5, main=&quot;Only top LDL SNPs&quot;, xlab=&quot;LDL effect&quot;, ylab=&quot;Height effect&quot;))</code></pre>
<p><img src="cad_ldl_files/figure-html/unnamed-chunk-10-3.png" width="672" /></p>
<p>It seems possible that, among top LDL SNPs, there is some negative correlation in the LDL and height effect sizes.</p>
</div>
<div id="sherlock-results" class="section level2">
<h2>Sherlock Results</h2>
<p>We pruned SNPs for LD within windows defined by Pickerell et al and downloaded from <a href="https://bitbucket.org/nygcresearch/ldetect-data/src/ac125e47bf7f/EUR/?at=master">(here</a>.</p>
<table style="width:100%;">
<colgroup>
<col width="9%" />
<col width="11%" />
<col width="16%" />
<col width="33%" />
<col width="18%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th>Test</th>
<th>LD Thresh</th>
<th><span class="math inline">\(\hat{\lambda}\)</span></th>
<th><span class="math inline">\(\hat{\lambda}/se(\hat{\lambda})\)</span></th>
<th>Corr (q-val &lt;0.1)</th>
<th>Corr test</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LDL-&gt;CAD</td>
<td>None (2.4m)</td>
<td>0.52</td>
<td>76.9</td>
<td>0.53</td>
<td>50.2</td>
</tr>
<tr class="even">
<td>LDL-&gt;CAD</td>
<td>0.5 (401k)</td>
<td>0.54</td>
<td>28.6</td>
<td>0.59</td>
<td>20.0</td>
</tr>
<tr class="odd">
<td>LDL-&gt;CAD*</td>
<td>0.5</td>
<td>0.50</td>
<td>15.8</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>LDL-&gt;CAD</td>
<td>0.1 (145k)</td>
<td>0.54</td>
<td>16.2</td>
<td>0.58</td>
<td>9.8</td>
</tr>
<tr class="odd">
<td>———</td>
<td>———–</td>
<td>——</td>
<td>——-</td>
<td>——</td>
<td>——</td>
</tr>
<tr class="even">
<td>CAD-&gt;LDL</td>
<td>None</td>
<td>0.09</td>
<td>45.0</td>
<td>0.08</td>
<td>2.93</td>
</tr>
<tr class="odd">
<td>CAD-&gt;LDL</td>
<td>0.5</td>
<td>0.09</td>
<td>14.8</td>
<td>0.08</td>
<td>0.73</td>
</tr>
<tr class="even">
<td>CAD-&gt;LDL*</td>
<td>0.5</td>
<td>0.04</td>
<td>4.3</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>CAD-&gt;LDL</td>
<td>0.1</td>
<td>0.09</td>
<td>7.8</td>
<td>-0.01</td>
<td>-0.06</td>
</tr>
<tr class="even">
<td>———</td>
<td>———–</td>
<td>——</td>
<td>——-</td>
<td>——</td>
<td>——</td>
</tr>
<tr class="odd">
<td>LDL-&gt;HT</td>
<td>None (2.5m)</td>
<td>-0.08</td>
<td>-27.2</td>
<td>-0.12</td>
<td>-9.3</td>
</tr>
<tr class="even">
<td>LDL-&gt;HT</td>
<td>0.5 (402k)</td>
<td>-0.04</td>
<td>-7.4</td>
<td>-0.19</td>
<td>-5.2</td>
</tr>
<tr class="odd">
<td>LDL-&gt;HT</td>
<td>0.1 (145k)</td>
<td>-0.04</td>
<td>-4.32</td>
<td>-0.14</td>
<td>-2.02</td>
</tr>
<tr class="even">
<td>———</td>
<td>———–</td>
<td>——</td>
<td>——-</td>
<td>——</td>
<td>——</td>
</tr>
<tr class="odd">
<td>HT-&gt;LDL</td>
<td>None</td>
<td>-0.10</td>
<td>-95.9</td>
<td>-0.045</td>
<td>-21.94</td>
</tr>
<tr class="even">
<td>HT-&gt;LDL</td>
<td>0.5</td>
<td>-0.09</td>
<td>-27.9</td>
<td>-0.05</td>
<td>-7.9</td>
</tr>
<tr class="odd">
<td>HT-&gt;LDL</td>
<td>0.1</td>
<td>-0.09</td>
<td>-15.15</td>
<td>-0.06</td>
<td>-5.2</td>
</tr>
</tbody>
</table>
</div>
<div id="estimating-g_21" class="section level2">
<h2>Estimating <span class="math inline">\(g_{2,1}\)</span></h2>
<p>We saw in simulations that having a bad estimate of <span class="math inline">\(g_{2,1}\)</span> can lead to inflated estimates of <span class="math inline">\(\lambda\)</span> and increased rates of false positives. I tried two alternative methods of estimating <span class="math inline">\(g_{2,1}\)</span>.</p>
<ul>
<li>Estimate <span class="math inline">\(g_{2,1}\)</span> using only SNPs that have a <span class="math inline">\(q\)</span>-value less than 0.2 for <strong>either</strong> of the traits considered. (so in the LDL-&gt; CAD analysis, we will estiamte <span class="math inline">\(g_{2,1}\)</span> by fitting ASH to the CAD effects of only SNPs that have either an LDL <span class="math inline">\(q\)</span>-value less than 0.2 or a CAD <span class="math inline">\(q\)</span>-value of less than 0.2 ) (qval<span class="math inline">\(&lt;0.2\)</span>)</li>
<li>Estimate <span class="math inline">\(g_{2,1}\)</span> using only SNPs that appear in the GWAS catalog. (gwas)</li>
<li>Estimate <span class="math inline">\(g_{2,1}\)</span> using only SNPs that have a <span class="math inline">\(q\)</span>-value less than 0.2 for the first trait. (qval-asc)</li>
</ul>
<p>These experiments are done using the set of SNPs pruned at a threshold of 0.1. Results are shown in the table below.</p>
</div>
<div id="accounting-for-confounding" class="section level2">
<h2>Accounting for confounding</h2>
<p>We observed that there is some sample overlap between the CAD and Height cohorts. Additionally, populations structure could lead to correlation between <span class="math inline">\(\hat{\beta}_1\)</span> and <span class="math inline">\(\hat{\beta}_2\)</span> conditional on the true effects <span class="math inline">\(\beta_1\)</span> and <span class="math inline">\(\beta_2\)</span>. To account for this, I added an additional parameter <span class="math inline">\(\rho\)</span> to account for this correlation. This extension of the model is described in <a href="sherlock2_compound_and_correlation.pdf">the notes</a> in Section 2.3. The correlation parameter <span class="math inline">\(\rho\)</span> is also estimated via maximum likelihood. Fitting this model in the four analyses discussed previously (LDL-&gt; CAD, CAD-&gt;LDL, LDL-&gt;HT, HT-&gt;LDL) in the SNP set pruned at an LD threshold of 0.1 gives the results in the table below.</p>
<table>
<thead>
<tr class="header">
<th>Test</th>
<th><span class="math inline">\(g_{2,1}\)</span></th>
<th><span class="math inline">\(\hat{\lambda}\)</span>(CI)</th>
<th><span class="math inline">\(\lambda\)</span>-stat</th>
<th><span class="math inline">\(\hat{\rho}\)</span>(CI)</th>
<th><span class="math inline">\(\rho\)</span>-stat</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LDL-&gt;CAD</td>
<td><span class="math inline">\(g_{2,0}\)</span></td>
<td>0.536(0.471, 0.600)</td>
<td>16.24</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>LDL-&gt;CAD</td>
<td>qval<span class="math inline">\(&lt;0.2\)</span></td>
<td>0.514(0.408, 0.619)</td>
<td>9.55</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>LDL-&gt;CAD</td>
<td>gwas</td>
<td>0.527(0.456, 0.596)</td>
<td>14.72</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>LDL-&gt;CAD</td>
<td><span class="math inline">\(g_{2,0}\)</span></td>
<td>0.51(0.448, 0.578)</td>
<td>15.55</td>
<td>0.024(0.018, 0.029)</td>
<td>8.60</td>
</tr>
<tr class="odd">
<td>LDL-&gt;CAD</td>
<td>qval<span class="math inline">\(&lt;0.2\)</span></td>
<td>0.480(0.372,0.589)</td>
<td>8.69</td>
<td>0.025(0.020, 0.031)</td>
<td>9.23</td>
</tr>
<tr class="even">
<td>LDL-&gt;CAD</td>
<td>gwas</td>
<td>0.501(0.432, 0.570)</td>
<td>14.14</td>
<td>0.024(0.019, 0.030)</td>
<td>8.77</td>
</tr>
<tr class="odd">
<td>LDL-&gt;CAD</td>
<td>qval-asc</td>
<td>0.491(0.404, 0.577)</td>
<td>11.12</td>
<td>0.025(0.019, 0.030)</td>
<td>8.97</td>
</tr>
<tr class="even">
<td>———</td>
<td>———-</td>
<td>———————–</td>
<td>——</td>
<td>———————-</td>
<td>——-</td>
</tr>
<tr class="odd">
<td>CAD-&gt;LDL</td>
<td><span class="math inline">\(g_{2,0}\)</span></td>
<td>0.089(0.067, 0.111)</td>
<td>7.76</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>CAD-&gt;LDL</td>
<td>qval<span class="math inline">\(&lt;0.2\)</span></td>
<td>0.049(0.005, 0.093)</td>
<td>2.18</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>CAD-&gt;LDL</td>
<td>gwas</td>
<td>0.075(0.054, 0.097)</td>
<td>6.97</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>CAD-&gt;LDL</td>
<td><span class="math inline">\(g_{2,0}\)</span></td>
<td>0.004 (-0.021, 0.030)</td>
<td>0.33</td>
<td>0.027(0.020, 0.034)</td>
<td>7.58</td>
</tr>
<tr class="odd">
<td>CAD-&gt;LDL</td>
<td>qval<span class="math inline">\(&lt;0.2\)</span></td>
<td>0.018(-0.029, 0.064)</td>
<td>0.78</td>
<td>0.028(0.022, 0.034)</td>
<td>8.82</td>
</tr>
<tr class="even">
<td>CAD-&gt;LDL</td>
<td>gwas</td>
<td>0.006(-0.019, 0.031)</td>
<td>0.51</td>
<td>0.026(0.019, 0.033)</td>
<td>7.62</td>
</tr>
<tr class="odd">
<td>CAD-&gt;LDL</td>
<td>qval-asc</td>
<td>0.004(-0.021, 0.029)</td>
<td>0.33</td>
<td>0.027(0.019, 0.034)</td>
<td>7.58</td>
</tr>
<tr class="even">
<td>———</td>
<td>———-</td>
<td>———————–</td>
<td>——</td>
<td>———————-</td>
<td>——-</td>
</tr>
<tr class="odd">
<td>LDL-&gt;HT</td>
<td><span class="math inline">\(g_{2,0}\)</span></td>
<td>-0.041(-0.059, -0.022)</td>
<td>-4.32</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>LDL-&gt;HT</td>
<td>qval<span class="math inline">\(&lt;0.2\)</span></td>
<td>-0.07(-0.113, -0.035)</td>
<td>-3.73</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>LDL-&gt;HT</td>
<td>gwas</td>
<td>-0.044(-0.066, -0.022)</td>
<td>-3.85</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>LDL-&gt;HT</td>
<td><span class="math inline">\(g_{2,0}\)</span></td>
<td>-0.028(-0.046, -0.009)</td>
<td>-2.94</td>
<td>-0.049(-0.056, -0.043)</td>
<td>-14.76</td>
</tr>
<tr class="odd">
<td>LDL-&gt;HT</td>
<td>qval<span class="math inline">\(&lt;0.2\)</span></td>
<td>-0.057(-0.096, -0.018)</td>
<td>-2.89</td>
<td>-0.049(-0.056, -0.042)</td>
<td>-14.65</td>
</tr>
<tr class="even">
<td>LDL-&gt;HT</td>
<td>gwas</td>
<td>-0.027 (-0.049,-0.006)</td>
<td>-2.46</td>
<td>-0.049(-0.056, -0.043)</td>
<td>-14.73</td>
</tr>
<tr class="odd">
<td>LDL-&gt;HT</td>
<td>qval-asc</td>
<td>-0.030(-0.054, -0.005)</td>
<td>-2.32</td>
<td>-0.049(-0.056, -0.043)</td>
<td>-14.81</td>
</tr>
<tr class="even">
<td>———</td>
<td>———-</td>
<td>———————–</td>
<td>——</td>
<td>———————-</td>
<td>——-</td>
</tr>
<tr class="odd">
<td>HT-&gt;LDL</td>
<td><span class="math inline">\(g_{2,0}\)</span></td>
<td>-0.092(-0.104, -0.080)</td>
<td>-15.15</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>HT-&gt;LDL</td>
<td>qval<span class="math inline">\(&lt;0.2\)</span></td>
<td>-0.079(-0.103, -0.055)</td>
<td>-6.49</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>HT-&gt;LDL</td>
<td>gwas</td>
<td>-0.086(-0.098, -0.073)</td>
<td>-13.68</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>HT-&gt;LDL</td>
<td><span class="math inline">\(g_{2,0}\)</span></td>
<td>-0.057(-0.072, -0.042)</td>
<td>-7.50</td>
<td>-0.032(-0.040, -0.023)</td>
<td>-7.60</td>
</tr>
<tr class="odd">
<td>HT-&gt;LDL</td>
<td>qval<span class="math inline">\(&lt;0.2\)</span></td>
<td>0.047 (-0.071, 0.022)</td>
<td>-3.73</td>
<td>-0.064(-0.076, -0.053)</td>
<td>-10.78</td>
</tr>
<tr class="even">
<td>HT-&gt;LDL</td>
<td>gwas</td>
<td>-0.050(-0.065, -0.035)</td>
<td>-6.51</td>
<td>-0.036(-0.044, -0.028)</td>
<td>-8.48</td>
</tr>
<tr class="odd">
<td>HT-&gt;LDL</td>
<td>qval-asc</td>
<td>-0.058(-0.073, -0.043)</td>
<td>-7.60</td>
<td>-0.031(-0.040, -0.023)</td>
<td>-7.57</td>
</tr>
</tbody>
</table>
</div>
<div id="permutations" class="section level2">
<h2>Permutations</h2>
<p>Since there was very little difference between estiates with <span class="math inline">\(g_{2,1} = g_{2,0}\)</span> and <span class="math inline">\(g_{2,1}\)</span> estimated using the <code>qval-asc</code> method, these permutations are run using <span class="math inline">\(g_{2,1} = g_{2,0}\)</span>. In each repetition, I permuted the effects and standard errors (together) for either height or CAD. I then re-ran Sherlock v4 to obtain an estimate of <span class="math inline">\(\lambda\)</span> and <span class="math inline">\(\rho\)</span>. Since this permutation scheme breaks the relationship between the two studies, we expect that <span class="math inline">\(E[\lambda_{perm}] = 0\)</span> and <span class="math inline">\(E[\rho_{perm}] = 0\)</span>.</p>
<p>Encouragingly - on permuted data, the statistics are approximately normally distributed.</p>
<pre class="r"><code>library(stats4)
perm &lt;- readRDS(&quot;../results/ht_ldl_v4.perm.RData&quot;)
r &lt;- readRDS(&quot;../results/ht_ldl.v4.RData&quot;)
hist(perm$lambda_stat, breaks=30, 
     main=&quot;Height-&gt;LDL permutation stats\n500 permutations&quot;, xlab=&quot;Statistic&quot;, 
     xlim=range(c(perm$lambda_stat), coef(r)[1]/sqrt(vcov(r)[1,1])),
     probability=TRUE)
abline(v=coef(r)[1]/sqrt(vcov(r)[1,1]), col=&quot;red&quot;)
curve(dnorm(x), add=TRUE)</code></pre>
<p><img src="cad_ldl_files/figure-html/perms_ht_ldl-1.png" width="672" /></p>
<pre class="r"><code>e &lt;- qnorm(p=seq(0, 1, length.out=502)[2:501])
plot(sort(perm$lambda_stat), e, xlab=&quot;Observed&quot;, ylab=&quot;Expected&quot;)
abline(0, 1)</code></pre>
<p><img src="cad_ldl_files/figure-html/perms_ht_ldl-2.png" width="672" /></p>
<pre class="r"><code>library(stats4)
perm &lt;- readRDS(&quot;../results/ldl_ht_v4.perm.RData&quot;)
r &lt;- readRDS(&quot;../results/ldl_ht.v4.RData&quot;)
st &lt;- coef(r)[1]/sqrt(vcov(r)[1,1])
hist(perm$lambda_stat, breaks=30, 
     main=&quot;LDL-&gt;HT permutation stats\n500 permutations&quot;, xlab=&quot;Statistic&quot;, 
     xlim=range(c(perm$lambda_stat), st),
     probability=TRUE)
abline(v=st, col=&quot;red&quot;)
curve(dnorm(x), add=TRUE)</code></pre>
<p><img src="cad_ldl_files/figure-html/perms_ldl_ht-1.png" width="672" /></p>
<pre class="r"><code>e &lt;- qnorm(p=seq(0, 1, length.out=502)[2:501])
plot(sort(perm$lambda_stat), e, xlab=&quot;Observed&quot;, ylab=&quot;Expected&quot;)
abline(0, 1)</code></pre>
<p><img src="cad_ldl_files/figure-html/perms_ldl_ht-2.png" width="672" /></p>
<pre class="r"><code>library(stats4)
perm &lt;- readRDS(&quot;../results/ldl_cad_v4.perm.RData&quot;)
r &lt;- readRDS(&quot;../results/ldl_cad.v4.RData&quot;)
st &lt;- coef(r)[1]/sqrt(vcov(r)[1,1])
hist(perm$lambda_stat, breaks=30, 
     main=&quot;LDL-&gt;CAD permutation stats\n500 permutations&quot;, xlab=&quot;Statistic&quot;, 
     xlim=range(c(perm$lambda_stat), st),
     probability=TRUE)
abline(v=st, col=&quot;red&quot;)
curve(dnorm(x), add=TRUE)</code></pre>
<p><img src="cad_ldl_files/figure-html/perms_ldl_cad-1.png" width="672" /></p>
<pre class="r"><code>e &lt;- qnorm(p=seq(0, 1, length.out=502)[2:501])
plot(sort(perm$lambda_stat), e, xlab=&quot;Observed&quot;, ylab=&quot;Expected&quot;)
abline(0, 1)</code></pre>
<p><img src="cad_ldl_files/figure-html/perms_ldl_cad-2.png" width="672" /></p>
<pre class="r"><code>library(stats4)
perm &lt;- readRDS(&quot;../results/cad_ldl_v4.perm.RData&quot;)
r &lt;- readRDS(&quot;../results/cad_ldl.v4.RData&quot;)
st &lt;- coef(r)[1]/sqrt(vcov(r)[1,1])
hist(perm$lambda_stat, breaks=30, 
     main=&quot;CAD-&gt;LDL permutation stats\n500 permutations&quot;, xlab=&quot;Statistic&quot;, 
     xlim=range(c(perm$lambda_stat), st),
     probability=TRUE)
abline(v=st, col=&quot;red&quot;)
curve(dnorm(x), add=TRUE)</code></pre>
<p><img src="cad_ldl_files/figure-html/perms_cad_ldl-1.png" width="672" /></p>
<pre class="r"><code>e &lt;- qnorm(p=seq(0, 1, length.out=502)[2:501])
plot(sort(perm$lambda_stat), e, xlab=&quot;Observed&quot;, ylab=&quot;Expected&quot;)
abline(0, 1)</code></pre>
<p><img src="cad_ldl_files/figure-html/perms_cad_ldl-2.png" width="672" /></p>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
