<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jean Morrison" />

<meta name="date" content="2018-04-11" />

<title>Crohn’s Disease and DGN Expression Data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sherlock2 model using Ash</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Crohn’s Disease and DGN Expression Data</h1>
<h4 class="author"><em>Jean Morrison</em></h4>
<h4 class="date"><em>April 11, 2018</em></h4>

</div>


<div id="data" class="section level2">
<h2>Data</h2>
<p>I applied CAUSE to look for genes whos expression levels could be causally associated with Crohn disease (CD). To do this I used GWAS sumarry statistics from the IBDgenetics consortium and eQTL summary statsitics calculated from the DGN data set.</p>
<div id="gwas-data" class="section level3">
<h3>GWAS Data</h3>
<p>The GWAS data are based on a meta analysis of 7 studies of European ancestry individuals. It includes 11,002,658 SNPs. After filtering out indels and strand ambiguous SNPs, we retain 8,126,897.</p>
</div>
<div id="eqtl-data" class="section level3">
<h3>eQTL Data</h3>
<p>eQTL test statisicts for the DGN data were calculated by Nick using 922 individuals. There are 12,441 transcripts and 625,883</p>
</div>
</div>
<div id="analysis-steps" class="section level2">
<h2>Analysis Steps</h2>
<div id="step-1-obtain-an-ld-pruned-set-of-snps-present-in-both-data-sets-snp-set-a" class="section level3">
<h3>Step 1: Obtain an LD pruned set of SNPs present in both data sets (SNP set A)</h3>
<p>This process takes about 10 minutes on 8 cores which is primarily for the LD pruning step. We LD prune to reduce the number of SNPs we feed into later steps. This leaves 607,428 SNPs before pruning and 112,073 SNPs after pruning. This pruning does not consider any of the association test statistics.</p>
</div>
<div id="step-2-run-ash-for-each-transcript-and-for-cd-using-snp-set-a" class="section level3">
<h3>Step 2: Run ASH for each transcript and for CD using SNP set A</h3>
<p>This takes about 1 minute per transcript.</p>
</div>
<div id="step-3-estimate-the-grid-mixture-parameter-using-snp-set-a-for-each-transcript." class="section level3">
<h3>Step 3: Estimate the grid mixture parameter using SNP set A for each transcript.</h3>
<p>This takes about 1 minute per transcript using 8 cores. We estimate both mixture parameters and .</p>
</div>
<div id="step-4-obtain-a-set-of-ld-pruned-snps-retaining-only-snps-with-eqtl-pvalue-1e-3-for-each-transcript.-snp-sets-b-i-for-i-in-1---12441" class="section level3">
<h3>Step 4: Obtain a set of LD pruned SNPs retaining only SNPs with eQTL pvalue &lt; 1e-3 for each transcript. (SNP sets B-i for i in 1 - 12441)</h3>
<p>This is the most computationally intensive step. We could consider not LD pruning and just using the whole LD block surrounding top SNPs. I ran this in 125 jobs (each processes 100 transcripts) each of which used 8 cores and 5G of memory. Each job took about 10 minutes.</p>
</div>
<div id="run-cause-for-each-transcript-using-snp-sets-b-i-i-in-1-to-124441" class="section level3">
<h3>Run CAUSE for each transcript using SNP sets B-i (i in 1 to 124441)</h3>
<p>This takes only a few seconds since SNP sets are usually fewer than 100.</p>
</div>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<p>The plot below shows the number of genes with a posterior median for <span class="math inline">\(q\)</span> above various thresholds.</p>
<pre class="r"><code>library(readr)
res &lt;- read_tsv(&quot;../data/cause_results_1e-3_summary.tsv.gz&quot;)</code></pre>
<pre><code>## Parsed with column specification:
## cols(
##   transcript = col_double(),
##   genename = col_character(),
##   p0_ash = col_double(),
##   q_0.05 = col_double(),
##   q_median = col_double(),
##   q_0.95 = col_double(),
##   b_0.05 = col_double(),
##   b_median = col_double(),
##   b_0.95 = col_double(),
##   zscore = col_double(),
##   file = col_character()
## )</code></pre>
<pre class="r"><code>x &lt;- seq(0, 1, length.out=100)
y &lt;- sapply(x, function(xx){sum(res$q_median &gt;= xx)})
plot(x, y, type=&quot;l&quot;, ylim=c(0, 200), ylab=&quot;Number of genes with median(q) &gt; x&quot;)</code></pre>
<p><img src="dgn_results_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
<p>There are 30 genes with posterior medians above 0.5.</p>
<p>Below I show a histogram of <span class="math inline">\(z\)</span>-scores comparing the causal and shared model.</p>
<pre class="r"><code>with(res, hist(zscore, breaks=100, probability = TRUE))
curve(dnorm(x, 0, 1), from=-4, to = 4, add=TRUE, lty=2)</code></pre>
<p><img src="dgn_results_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<p>These <span class="math inline">\(z\)</span>-scores don’t appear to be normally distributed which suggests they may not actually behave as <span class="math inline">\(z\)</span>-scores. I will check their distribution in simulations.</p>
<p>Below I show 90% credible intervals and posterior medians for q for all genes with a posterior median greater than 0.5.</p>
<pre class="r"><code>gene_names &lt;- arrange(res, q_median) %&gt;% with(., genename)
res$genename &lt;- factor(res$genename, levels=gene_names)

plt &lt;- filter(res, q_median &gt; 0.5) %&gt;%
        ggplot(.) + geom_point(aes(x=genename, y=q_median)) +
        geom_errorbar(aes(x=genename, ymin=q_0.05, ymax=q_0.95)) +
        scale_y_continuous(limits=c(0, 1)) +
        theme_bw() + theme(axis.text.x=element_text(angle=90)) 
plt</code></pre>
<p><img src="dgn_results_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<p>Besided the top two genes, I couldn’t find any relationship between most of these genes and Crohn disease. The exceptions are</p>
<ul>
<li>SLC22A5: Several Crohn’s related variants; Carnitine deficiency, systemic primary</li>
<li>C1QTNF4: (also called CTRP4) May act as a pro-inflammatory cytokine, that induces the activation of NF-kappa-B signaling pathway and up-regulates IL6 production. Implicated in Systemic Lupus erythematosus (Pullabhatla et al Hum. Mol. Gen. 2018)</li>
</ul>
<p>C1QTNF4 has not (that I can find) been linked to Crohn disease but it has been linked to SLE which is related. It also appears that many of these may be correlated with each other. To check this, I ran CAUSE on pairwise combinations of the top genes:</p>
<p><img src="dgn_matrix.png" width="500"></p>
<p>In particular, C1QTNF4 is correlated with four of the top genes suggesting that these results may be spurious.</p>
<p>Notes about other genes:</p>
<ul>
<li>BLK: This gene encodes a nonreceptor tyrosine-kinase of the src family of proto-oncogenes that are typically involved in cell proliferation and differentiation. The protein has a role in B-cell receptor signaling and B-cell development. The protein also stimulates insulin synthesis and secretion in response to glucose and enhances the expression of several pancreatic beta-cell transcription factors. [provided by RefSeq, Aug 2010] Diseases associated with BLK include Maturity-Onset Diabetes Of The Young, Type 11 and Maturity-Onset Diabetes Of The Young. Among its related pathways are Apoptosis Modulation and Signaling and Innate Immune System. Gene Ontology (GO) annotations related to this gene include transferase activity, transferring phosphorus-containing groups and protein tyrosine kinase activity. An important paralog of this gene is HCK.</li>
</ul>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
